test/src/unittests/rhythm/test_loopbpmconfidence.py

```diff
@@ -0,0 +1,212 @@
+        confidence = LoopBpmConfidence()(audio[0:len90], bpmEstimate)              
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.01) 
+
+        bpmEstimate = 120
+        expectedConfidence = 0.43 # rounded from measured 0.4305669069290161
+        confidence = LoopBpmConfidence()(audio[0:len90], bpmEstimate)                          
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.01) 
+        
+        bpmEstimate = 70
+        expectedConfidence = 0.50 # rounded from measured 0.5011640191078186
+        confidence = LoopBpmConfidence()(audio[0:len90], bpmEstimate)                     
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.01)
+        
+        # Truncated Audios 90% and 1% at beginning
+        bpmEstimate = 125
+        expectedConfidence = 0.92 # rounded from measured 0.9199735522270203
+        confidence = LoopBpmConfidence()(audio[len01:len90], bpmEstimate)                         
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.01) 
+
+        bpmEstimate = 120
+        expectedConfidence =  0.36 # rounded from measured 0.3631746172904968
+        confidence = LoopBpmConfidence()(audio[len01:len90], bpmEstimate)           
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.01) 
+        
+        bpmEstimate = 70 
+        expectedConfidence =  0.79 # rounded from measured 0.7951852083206177
+        confidence = LoopBpmConfidence()(audio[len01:len90], bpmEstimate)
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.01) 
+
+
+    """
+    The confidence returned is based on comparing the duration of the audio signal with 
+    multiples of the BPM estimate. 
+    The algorithm considers edge cases in which the signal includes non-musical 
+    silence at the start or end of the audio signal (or both) and returns the confidence for the best fit.
+    For more details refer to paper: https://repositori.upf.edu/handle/10230/33113
+
+    Additional Tests were made with following loop
+    "458538__flowerpunkchip__mellifluousanonymous-120bpm-loop-youareevil.mp3'"
+    https://freesound.org/people/Flowerpunkchip/sounds/458538/
+
+    Front end slience Confidence is  0.9915646314620972
+    Back end silence Confidence  is  0.9915646314620972
+    Silence Both end confidence is  0.9915646314620972    
+    """
+    def testSilentEdges(self):
+
+        audio = MonoLoader(filename=join(testdata.audio_dir, 'recorded', 'techno_loop.wav'))() 
+        bpmEstimate = 125
+        confidence = LoopBpmConfidence()(audio, bpmEstimate)              
+        lenSilence = int(0.01*len(audio)) # Choose a silent length 1% of audio
+        silentAudio = zeros(lenSilence)
+
+        # case 1: there is non-musical* silence before the loop starts
+        signal1  = numpy.append(silentAudio, audio)
+        confidence = LoopBpmConfidence()(signal1, bpmEstimate)              
+        self.assertGreater(confidence, 0.9) 
```

Comment:
I am trying to understand this test. What are the 0.9 and 0.27 values and what is reasoning for using `assertGreater`?

Reply:
values have been updated with new GT.
The reasoning on assertGreater is to assure measured confidence is  greater than a benchmark that is reasonably high (0.8 to 0.9)